### DISCUSSION TITLE
Last item of an array as default content of multiple editors

### DISCUSSION THREAD

**Discussion by eakl** [ueberdosis/tiptap#2613] [General Questions]
Hi, I want to initialize the editor with a JSON object as follows:
1. `<Editor content={'Initial text'} />` component receive a content props (as a string)
2. The Editor component transforms the text in a JSON object
3. The JSON object serves a the initial Editor content.

Problem is when I loop through an array of contents and render as many editor as contents, all the editors have the last item of the array as the default content.

The issue is reproduced here: https://codesandbox.io/s/frosty-yalow-19k5u7?file=/src/Item.tsx
- You can see that the data array has different content but all editor are render with the same content (last item of the array).

#### Initial JSON
```javascript
export default {
  type: "doc",
  content: [
    {
      type: "paragraph",
      content: [
        {
          type: "text",
          text: ""
        }
      ]
    }
  ]
};

```

#### This doesn't work
```javascript
export const Editor = ({ content }: any) => {
  const defaultContent = { ...DEFAULT };
  defaultContent.content[0].content[0].text = content; // text key is replaced with content
  const data = useRef<Record<string, any>>(defaultContent); // I need to do this to keep track of the last content before edit

  const editor = useEditor({
    extensions: [StarterKit],
    content: data.current
  });

  return <EditorContent editor={editor} />;
};
```

#### This works
```javascript
export const Editor = ({ content }: any) => {
  const data = useRef<Record<string, any>>(content);

  const editor = useEditor({
    extensions: [StarterKit],
    content: data.current
  });

  return <EditorContent editor={editor} />;
};
```

**Comment by philippkuehn** [comment]
You have to create a deep clone for `defaultContent`. Try something like this:

```js
const defaultContent = JSON.parse(JSON.stringify(DEFAULT))
```
